### PR TITLE
Cross-chain sync resolution - `Voter.ts` contract on `DistributeReward` event

### PR DESCRIPTION
**Pull request by FonsSicca**
Implements:
- Similar resolution workflow as that in #542 but for `DistributeReward` event

## Summary by CodeRabbit

* **New Features**
  * Cross-chain reward distribution support with schema updates for root-gauge→root-pool mapping and deferred pending distributions.
* **Refactor**
  * Unified, resilient flushing of pending votes and reward distributions when mappings become available.
* **Tools**
  * Added safe async error-logging and block/log-index sorting helpers to improve processing reliability.
* **Tests**
  * Expanded end-to-end and unit tests for cross-chain flows, deferred processing, and edge cases.

### CODE REVIEW

File: src/EventHandlers/Voter/CrossChainPendingResolution.ts [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+          pending,
+          leafPoolAddress,
+          leafChainId,
+        ),
+      deleteProcessedPendingDistribution,
+      CrossChainPendingResolutionLogPrefix.Distributions,
+      pending.id,
+    );
+  }
+}
```
Comment (FonsSicca) [inline src/EventHandlers/Voter/CrossChainPendingResolution.ts]: this is actually the net new logic since pending votes logic was already in the previous pr
